In [1]:
from scipy import stats
import numpy as np
import math

In [3]:
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
data

[1.13,
 1.55,
 1.43,
 0.92,
 1.25,
 1.36,
 1.32,
 0.85,
 1.07,
 1.48,
 1.2,
 1.33,
 1.18,
 1.22,
 1.29]

## a. Build 99% Confidence Interval Using Sample Standard Deviation
Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.


In [24]:
# Sample size
n = 15 # from given data

# Sample mean
x_bar = np.mean(data)

# Sample standard deviation
s = np.std(data, ddof=1)

# Confidence level
confidence = 0.99

# Degrees of freedom
df = n - 1

# t critical value
t_value = stats.t.ppf(1 - (1-confidence)/2, df) # df is degree of freedom used for t-test-distribution n-1(15-1=14)

# Standard error
se = s / math.sqrt(n) # reduces variability

# Margin of error
me = t_value * se # shows extend of interval from mean

# Confidence interval  using upper lower range cause we don't know exact population mean and sample mean varies sample to sample so it is appropriate way to represent interval in the form of range
lower = mean - me
upper = mean + me

# Output
print("Sample Mean:", x_bar)
print("Sample Standard Deviation:", s)
print("t-value:", t_value)
print("Margin of Error:", me)

print("\n99% Confidence Interval:")
print((lower, upper))

Sample Mean: 1.2386666666666666
Sample Standard Deviation: 0.19316412956959936
t-value: 2.976842734370834
Margin of Error: 0.14846932822817596

99% Confidence Interval:
(np.float64(1.0901973384384906), np.float64(1.3871359948948425))


### We are 99% confident true mean durability lies between 1.09 and 1.39 million characters.

## b. Build 99% Confidence Interval Using Known Population Standard Deviation
If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.



In [23]:
# Known population standard deviation
sigma = 0.2

# Sample size
n = 15

# Sample mean
x_bar = np.mean(data)

# Confidence level
confidence = 0.99

# z critical value
z_value = stats.norm.ppf(1 - (1-confidence)/2)

# Standard error
se = sigma / math.sqrt(n)

# Margin of error
me = z_value * se

# Confidence interval
lower = mean - me
upper = mean + me

# Output
print("Sample Mean:", x_bar)
print("Population Standard Deviation:", sigma)
print("z-value:", z_value)
print("Margin of Error:", me)

print("\n99% Confidence Interval:")
print((lower, upper))

Sample Mean: 1.2386666666666666
Population Standard Deviation: 0.2
z-value: 2.5758293035489004
Margin of Error: 0.13301525327090588

99% Confidence Interval:
(np.float64(1.1056514133957607), np.float64(1.3716819199375725))


### We are 99% confident that the true average durability lies between 1.106 and 1.372 million characters.